src/test/java/io/github/floverfelt/find/and/replace/maven/plugin/FindAndReplaceMojoTest.java

```diff
@@ -504,23 +520,18 @@ public void testEverything() throws IOException, NoSuchFieldException, IllegalAc
   }
```

Comment:
You'll need to add some tests which verify the false condition of replaceAll not just the true. At minimum you should add tests for:
1. A file name with multiple matches and replaceAll set to false. Verify that only the first is replaced.
2. A directory name with multiple matches and replaceAll set to false. Verify that only the first is replaced.
3. A file with multiple matches and replaceAll set to false. Verify that only the first is replaced.

Reply:
@PeterGit96 added a comprehensive batch of tests

Reply:
Add junit tests to verify that only the first (filename, directory name, file contents) is replaced with replaceAll set to false; 

FILENAME
A file name with multiple matches and replaceAll set to false. Verify that only the first is replaced.

testFilenamesReplaceFirst
testFilenamesRecursiveReplaceFirst
testFilenamesRecursiveFileMasksReplaceFirst
testFilenamesRecursiveFileMasksWithExclusionsReplaceFirst

DIRECTORY NAME
A directory name with multiple matches and replaceAll set to false. Verify that only the first is replaced.

testDirectoryNamesReplaceFirst
testDirectoryNamesRecursiveReplaceFirst
testDirectoryNamesRecursiveExclusionsReplaceFirst

FILE CONTENTS
A file with multiple matches and replaceAll set to false. Verify that only the first is replaced.

testFileContentsReplaceFirst
testFileContentsNonStandardEncodingFindRegexReplaceFirst
testFileContentsNonStandardReplaceValueReplaceFirst
testFileContentsNonStandardFindAndReplaceValueReplaceFirst
testFileContentsRecursiveReplaceFirst
testFileContentsNotRecursiveReplaceFirst
testFileContentsRecursiveFileMasksReplaceFirst
testFileContentsRecursiveFileMasksExclusionsReplaceFirst
testEverythingReplaceFirst

Reply:
Thank you!